### PR TITLE
[bashible] fix bb-event-error-create

### PR DESCRIPTION
**Pull request by RomanenkoDenys**
Signed-off-by: Denis Romanenko <denis.romanenko@flant.com>

## Description

Fix error in bb-event-error-create bashible function.

## Why do we need it, and what problem does it solve?

1. When bashible tries to create an event, we get an error because we do not have access from system:nodes group to events.k8s.io api.
2. If cluster in bootstrapping state, we cannot have working API to send events, so we simply suppress event creating errors to prevent bashible fail. 
## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: candi
type: fix
summary: Fix error in bb-event-error-create bashible function.
impact_level: low
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/gcp'...

File: candi/bashible/bashible.sh.tpl [hunk 1]
```diff
@@ -27,10 +27,11 @@ function bb-event-error-create() {
     # underscore with dash due to regexp.
     # All of stderr outputs are stored in the eventLog file.
     # step is used as argument for function call.
+    # If event creation failed, error from kubectl suppressed.
     step="$1"
     eventName="$(echo -n "$(hostname -s)")-$(echo $step | sed 's#.*/##; s/_/-/g')"
     eventLog="/var/lib/bashible/step.log"
-    kubectl apply -f - <<EOF
+    kubectl_exec apply -f - <<EOF || true
```
Comment (yalosev) [inline candi/bashible/bashible.sh.tpl:34]: should not we also move `stderr` to `/dev/null` ?

Comment (RomanenkoDenys) [reply candi/bashible/bashible.sh.tpl:34]: no because if kubectl exit code > 0 bashible fails

**Review by yalosev — DISMISSED** [review]

File: testing/matrix/linter/rules/roles/placement.go [hunk 2]
```diff
@@ -316,7 +316,7 @@ func handleNestedRBACForUs(m utils.Module, object storage.StoreObject, shortPath
 
 	switch {
 	case strings.HasPrefix(objectName, localPrefix):
-		if namespace != m.Namespace {
+		if namespace != m.Namespace && namespace != "default" {
```
Comment (nabokihms) [inline testing/matrix/linter/rules/roles/placement.go:319]: Oh no 😞

Comment (nabokihms) [reply testing/matrix/linter/rules/roles/placement.go:319]: This condition must not be changed. `default` is considered a system namespace, so the rule's name must start with `d8:` prefix.

File: modules/040-node-manager/templates/bashible/rbac-for-us.yaml [hunk 3]
```diff
@@ -52,6 +52,20 @@ rules:
...
+metadata:
+  name: bashible-events
+  namespace: default
+  {{- include "helm_lib_module_labels" (list . ) | nindent 2 }}
+rules:
+  - apiGroups: ["events.k8s.io"]
+    resources:
+      - events
+    verbs:
+      - "*"
```
Comment (nabokihms) [inline modules/040-node-manager/templates/bashible/rbac-for-us.yaml:67]: Too broad, consider specifying only required verbs

File: candi/bashible/bashible.sh.tpl [hunk 4]
```diff
@@ -41,7 +42,7 @@ function bb-event-error-create() {
           apiVersion: v1
           kind: Node
           name: '$(hostname -s)'
-          uid: "$(kubectl get node $(hostname -s) -o jsonpath='{.metadata.uid}')"
+          uid: "$(kubectl_exec get node $(hostname -s) -o jsonpath='{.metadata.uid}')"
```
Comment (nabokihms) [inline candi/bashible/bashible.sh.tpl:45]: BTW, uid is not required and can be avoided.

**Review by nabokihms — CHANGES_REQUESTED** [review]
Some questions regarding the event error create function:
1. The reason field is always `Failed`. `Failed` is too abstract. I expected to see an instruction on what should I do next, or at least, what is broken.

2. The action field is always equal to `Binding`. I assume it should be something like `BashibleFailed`.
> action is what action was taken/failed regarding to the regarding object. It is machine-readable. This field cannot be empty for new Events and it can have at most 128 characters.

What is the meaning of the `Binding`?

3. Why is there the `annotations.timestamp` field? It seems like eventTime is enough.

File: testing/matrix/linter/rules/roles/placement.go (hunk 2 above)
Comment (apolovov) [reply testing/matrix/linter/rules/roles/placement.go:319]: btw, it is better to use v1.NamespaceDefault.

File: candi/bashible/bashible.sh.tpl (hunk 4 above)
Comment (yalosev) [reply candi/bashible/bashible.sh.tpl:45]: Nope. Without uid event is not binded to the object. I mean when you make `kubectl describe node X` - you will not fetch an event without id

File: testing/matrix/linter/rules/roles/placement.go (hunk 2 above)
Comment (RomanenkoDenys) [reply testing/matrix/linter/rules/roles/placement.go]: fixed by proper way

**Comment by RomanenkoDenys** [comment]
> Some questions regarding the event error create function:
> 
> 1. The reason field is always `Failed`. `Failed` is too abstract. I expected to see an instruction on what should I do next, or at least, what is broken.

changed to BashibleStepFailed

> 2. The action field is always equal to `Binding`. I assume it should be something like `BashibleFailed`.
> 
> > action is what action was taken/failed regarding to the regarding object. It is machine-readable. This field cannot be empty for new Events and it can have at most 128 characters.
> 
> What is the meaning of the `Binding`?

changed to BashibleStepExecution

> 
> 3. Why is there the `annotations.timestamp` field? It seems like eventTime is enough.

removed

File: modules/040-node-manager/templates/bashible/rbac-for-us.yaml (hunk 3 above)
Comment (RomanenkoDenys) [reply modules/040-node-manager/templates/bashible/rbac-for-us.yaml]: changed to get, list, create, update, patch

**Review by yalosev — APPROVED** [review]

**Review by nabokihms — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/aws'...

**Review by apolovov — APPROVED** [review]

**Pull request merged**